Kiosk/Auction Listings/ListingsViewModel.swift

```diff
@@ -90,27 +91,36 @@ class ListingsViewModel: NSObject, ListingsViewModelType {
             .bindTo(saleArtworks)
             .addDisposableTo(rx_disposeBag)
 
-        showSpinnerSignal = saleArtworks.map { saleArtworks in
-            return saleArtworks.count == 0
+        showSpinnerSignal = sortedSaleArtworks.map { sortedSaleArtworks in
+            return sortedSaleArtworks.count == 0
         }
 
         gridSelectedSignal = selectedIndexSignal.map { ListingsViewModel.SwitchValues(rawValue: $0) == .Some(.Grid) }
 
-        let distinctSaleArtworks: Observable<[SaleArtwork]> = saleArtworks
+        let distinctSaleArtworks = saleArtworks
             .asObservable()
             .distinctUntilChanged { (lhs, rhs) -> Bool in
                 return lhs == rhs
             }
+            .mapReplace(0)
```

Comment:
It's to satisfy the type system; combineLatest/reduce can only be applied to an array of observables of the same type. I'll add a comment.